django/db/models/query.py

```diff
@@ -338,7 +338,22 @@ def __or__(self, other):
     ####################################
 
     def _iterator(self, use_chunked_fetch, chunk_size):
-        yield from self._iterable_class(self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size)
+        clone = self._chain()
```

Comment:
It'd be great if we could avoid this and all of the following `if not self._prefetch_related_lookups`

e.g.

```python
# Avoid materialization of chunks of objects if no prefetching must take
# place.
if not self._prefetch_related_lookups:
    yield from self._iterable_class(self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size)
    return
```

That'll make sure the exact same behavior is preserved when no prefetching is involved and allow you to revert the `test_server_side_cursors.py` changes.

Reply:
Yep, makes sense. I've reverted the test changes.